File: backend/models/Income.js
// models/Income.js
import mongoose from 'mongoose';

const IncomeSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId, required: true, ref: 'User' },
  amount: { type: Number, required: true },
  source: { type: String, required: true }, // e.g., Salary, Freelance, Other
  description: { type: String },
  date: { type: Date, default: Date.now }
});

export default mongoose.model('Income', IncomeSchema);

File: frontend/expensetracker/lib/models/income.dart
class Income {
  final String id;
  final String userId;
  final double amount;
  final String source;
  final String description;
  final String date;

  Income({
    required this.id,
    required this.userId,
    required this.amount,
    required this.source,
    required this.description,
    required this.date,
  });

  factory Income.fromJson(Map<String, dynamic> json) {
    return Income(
      id: json['_id'],
      userId: json['user_id'],
      amount: json['amount'].toDouble(),
      source: json['source'],
      description: json['description'],
      date: json['date'],
    );
  }
}
